tools/lambda-promtail/main.tf

```diff
@@ -0,0 +1,83 @@
+provider "aws" {
+  region = "us-east-2"
+}
+
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name = "iam_for_lambda"
+
+  assume_role_policy = jsonencode({
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+          "Service": "lambda.amazonaws.com"
+        },
+        "Effect": "Allow",
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "logs" {
+  name = "lambda-logs"
+  role = aws_iam_role.iam_for_lambda.name
+  policy = jsonencode({
+    "Statement" : [
+      {
+        "Action" : [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+        ],
+        "Effect" : "Allow",
+        "Resource" : "arn:aws:logs:*:*:*",
+      }
+    ]
+  })
+}
+
+resource "aws_lambda_function" "lambda_promtail" {
+  image_uri     = "your-image-ecr:latest"
+  function_name = "lambda_promtail"
+  role          = aws_iam_role.iam_for_lambda.arn
+
+  timeout      = 60
+  memory_size  = 128
+  package_type = "Image"
+
+  # vpc_config {
```

Comment:
Can we expose this in a conditional configuration rather than via comments?

Reply:
I'll have a look at terraform docs for doing this.